core/src/main/java/com/squarespace/template/plugins/platform/CommerceUtils.java

```diff
@@ -140,6 +140,78 @@ public static JsonNode getLowestPriceAmongVariants(JsonNode item) {
     }
   }
 
+  public static JsonNode getPricingOptionsAmongLowestVariant(JsonNode item) {
+    ProductType type = getProductType(item);
+    JsonNode structuredContent = item.path("structuredContent");
+
+    switch (type) {
+      case PHYSICAL:
+      case SERVICE:
+        JsonNode variants = structuredContent.path("variants");
+        if (variants.size() == 0) {
+          return null;
+        }
+
+        JsonNode first = variants.get(0);
+        JsonNode moneyNode = isTruthy(first.path("onSale"))
+                ? first.path("salePriceMoney")
+                : first.path("priceMoney");
+
+        JsonNode pricingOptions = first.path("pricingOptions");
+
+        Decimal price = getAmountFromMoneyNode(moneyNode);
+
+        for (int i = 1; i < variants.size(); i++) {
+          JsonNode var = variants.get(i);
+          JsonNode currentMoneyNode = isTruthy(var.path("onSale"))
+                  ? var.path("salePriceMoney")
+                  : var.path("priceMoney");
+
+          Decimal current = getAmountFromMoneyNode(currentMoneyNode);
+          if (current.compare(price) < 0) {
+            price = current;
+            pricingOptions = var.path("pricingOptions");
+          }
+        }
+
+        return pricingOptions;
+
+      default:
+        return null;
+    }
+  }
+
+  public static JsonNode getSubscriptionMoneyFromPricingOptions(JsonNode pricingOptions, String subscriptionId) {
+    JsonNode chosenSubscription = null;
+
+    if (pricingOptions == null || pricingOptions.size() == 0) {
+      return DEFAULT_MONEY_NODE;
+    }
+
+    if (subscriptionId == null) {
```

Comment:
In what scenario would we get a null subscriptionId and a non-null set of pricingOptions?